x-pack/plugins/ml/public/application/jobs/jobs_list/components/reset_job_modal/reset_job_modal.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiSpacer,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiLoadingSpinner,
+  EuiText,
+} from '@elastic/eui';
+
+import { resetJobs } from '../utils';
+import { RESETTING_JOBS_REFRESH_INTERVAL_MS } from '../../../../../../common/constants/jobs_list';
+import type { MlSummaryJob } from '../../../../../../common/types/anomaly_detection_jobs';
+import { OpenJobsWarningCallout } from './open_jobs_warning_callout';
+
+type ShowFunc = (jobs: MlSummaryJob[]) => void;
+
+interface Props {
+  setShowFunction(showFunc: ShowFunc): void;
+  unsetShowFunction(): void;
+  refreshJobs(): void;
+}
+
+export const ResetJobModal: FC<Props> = ({ setShowFunction, unsetShowFunction, refreshJobs }) => {
+  const [resetting, setResetting] = useState(false);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [jobIds, setJobIds] = useState<string[]>([]);
+  const [jobs, setJobs] = useState<MlSummaryJob[]>([]);
+
+  useEffect(() => {
+    if (typeof setShowFunction === 'function') {
+      setShowFunction(showModal);
+    }
+    return () => {
+      if (typeof unsetShowFunction === 'function') {
+        unsetShowFunction();
+      }
+    };
+  }, []);
+
+  const showModal = useCallback((tempJobs: MlSummaryJob[]) => {
+    setJobIds(tempJobs.map(({ id }) => id));
+    setJobs(tempJobs);
+    setModalVisible(true);
+    setResetting(false);
+  }, []);
+
+  const closeModal = useCallback(() => {
+    setModalVisible(false);
+  }, []);
+
+  const resetJob = useCallback(async () => {
+    setResetting(true);
+    await resetJobs(jobIds);
+    closeModal();
+
+    setTimeout(() => {
+      refreshJobs();
+    }, RESETTING_JOBS_REFRESH_INTERVAL_MS);
```

Comment:
this part looks alarming, I reckon we should not rely on the hardcoded timeout. in the cloud instances, requests tend to take longer

Reply:
This is also how the delete modal works, it delays for a couple of seconds just in case the job states haven't caught up. The jobs list then starts a poll to track the real state of reverting and deleting processes.

However.... one key difference between this modal and the delete job modal is that there is an `await` on revert request, meaning that this 2s delay isn't needed. The revert has already been assigned a task ID by this point.
I'll remove this timeout.

Reply:
Updated in https://github.com/elastic/kibana/pull/108039/commits/06f04dc840fb5a6cf2d98bd62bf8fa7a77f9b747

Reply:
Adding this delay back in with https://github.com/elastic/kibana/pull/108039/commits/115ad46b9089490f9881de22c1b33da590dc3901
We need to give elasticsearch a second for the blocked property to appear in the job.

Reply:
What happens in case of a severe network throttling? Maybe we can define and observable for jobs, and on each update look up for the blocked property?

Reply:
if the network is very slow, i can't see it making a difference. the call to reset is on an await, so this timeout won't start until the call to reset has returned.
Waiting a second to then refresh the jobs list is just there to give the task manager a bit of a breather before we ask if any tasks are running.

> Maybe we can define and observable for jobs, and on each update look up for the blocked property?

This is what is coming in the next PR. It's what we already do for "deleting" jobs, but that needs updating to use the new `blocked` property instead.